js/plugin/connection.ts

```diff
@@ -27,7 +28,30 @@ export const ConnectionMixin = (SuperClass) => {
       this.dispatchEvent(new CustomEvent(event, { detail, bubbles: true }));
     }
 
+    private onConnectionReady = () => {
+      this.connected = true;
+      this.fireEvent("browser-mod-connected");
+    }
+
+    private onConnectionLost = () => {
+      this.connected = false;
+      this.ready = false;
+      this.fireEvent("browser-mod-disconnected");
+    }
+
     private incoming_message(msg) {
+      // Check for connection state
+      if (!this.connected) {
+        this.onConnectionReady();
+      }
```

Comment:
I moved this away from `update_config()` to keep that method's responsibility clear.

Instead `incoming_message()` became "a hub" of letting the plugin know that it is well-connected to it's component counterpart.